semcore/dropdown-menu/src/DropdownMenu.jsx

```diff
@@ -387,22 +387,23 @@ class DropdownMenuRoot extends Component {
   }
 
   componentDidUpdate(prevProps) {
-    if (this.asProps.visible !== prevProps.visible) {
+    if (this.asProps.visible !== prevProps.visible && prevProps.visible !== undefined) {
       if (!this.asProps.visible) {
         this.handlers.highlightedIndex(null);
         this.highlightedItemRef.current = null;
         this.ignoreTriggerKeyboardFocusUntil = Date.now() + 100;
         if (document.activeElement === document.body || isFocusInside(this.popperRef.current)) {
           setFocus(this.triggerRef.current);
         }
-      } else {
-        setTimeout(() => {
-          const selectedItemIndex = this.itemProps.findIndex((item) => item.selected);
-          if (selectedItemIndex === -1 || this.asProps.highlightedIndex !== null) return;
-          this.handlers.highlightedIndex(selectedItemIndex);
-        }, 0);
       }
     }
+    if (this.asProps.visible !== prevProps.visible) {
```

Comment:
I'm not sure that we should highlight selected item on each change of visible prop, because in previous logic we did it only for `visible=true`